ouroboros-network-api/src/Ouroboros/Network/NodeToNode/Version.hs

```diff
@@ -196,42 +185,7 @@ nodeToNodeCodecCBORTerm version
         decodeTerm t
           = Left $ T.pack $ "unknown encoding: " ++ show t
      in CodecCBORTerm { encodeTerm, decodeTerm }
-  | version >= NodeToNodeV_11
-  , version <= NodeToNodeV_12 =
-    let encodeTerm :: NodeToNodeVersionData -> CBOR.Term
-        encodeTerm NodeToNodeVersionData { networkMagic, diffusionMode, query }
-          = CBOR.TList
-              [ CBOR.TInt (fromIntegral $ unNetworkMagic networkMagic)
-              , CBOR.TBool (case diffusionMode of
-                             InitiatorOnlyDiffusionMode         -> True
-                             InitiatorAndResponderDiffusionMode -> False)
-                              -- There's a bug in this versions where the
-                              -- agreed PeerSharing value on the remote side
-                              -- is whatever that got proposed, so we have to
-                              -- disable peer sharing with such nodes to avoid
-                              -- protocol violations
-              , CBOR.TInt 0 -- 0 corresponds to PeerSharingDisabled
-              , CBOR.TBool query
-              ]
 
-        decodeTerm :: CBOR.Term -> Either Text NodeToNodeVersionData
-        decodeTerm (CBOR.TList [CBOR.TInt x, CBOR.TBool diffusionMode, CBOR.TInt _, CBOR.TBool query])
-          | x >= 0
-          , x <= 0xffffffff
-          = Right
-              NodeToNodeVersionData {
-                  networkMagic = NetworkMagic (fromIntegral x),
-                  diffusionMode = if diffusionMode
-                                  then InitiatorOnlyDiffusionMode
-                                  else InitiatorAndResponderDiffusionMode,
-                  peerSharing = PeerSharingDisabled,
-                  query = query
-                }
-          | x < 0 || x > 0xffffffff
-          = Left $ T.pack $ "networkMagic out of bound: " <> show x
-        decodeTerm t
-          = Left $ T.pack $ "unknown encoding: " ++ show t
-     in CodecCBORTerm { encodeTerm, decodeTerm }
```

Comment:
When you remove this code the `otherwise` code below it breaks. 
We don't know how to encode or decode any versions older than `NodeToNodeV_13` with your change.

You should remove the `otherwise` case, and if it GHC complains the `otherwise`case should be an error.